facades/PC/build.gradle.kts

```diff
@@ -30,396 +28,370 @@ dateTimeFormat.timeZone = TimeZone.getTimeZone("UTC")
+    from("launchScripts") {
+        exclude("TeraEd.exe")
     }
 
     from("$buildDir/$versionFileName") {}
 
     into(subDirLibs) {
-        from configurations.runtimeClasspath
-        from project(':engine').jar
-        from ("$buildDir/libs") {
-            include "*.jar"
+        from(configurations.runtimeClasspath)
+        from(tasks.getByPath(":engine:jar"))
+        from("$buildDir/libs") {
+            include("*.jar")
             rename {
-                'Terasology.jar'
+                "Terasology.jar"
             }
         }
     }
     into(dirNatives) {
-        from "$rootDir/$dirNatives"
+        from("$rootDir/$dirNatives")
     }
 }
 
 // Distribute modules - only grabs Core in Jenkins but locally will grab any present. "Distros" now handle Jenkins packs
-task distModules (type: Sync) {
+tasks.register<Sync>("distModules") {
     description = "Prepares local modules for distribution"
-    dependsOn distApp
-    dependsOn rootProject.moduleJars
+    dependsOn("distApp")
+    dependsOn(":moduleJars")
 
     // So this is probably a hack, but it works ;-) It does not work if it is in distApp, default "into" quirk ?
     into("${distsDirectory.get().asFile}/app/modules")
-    rootProject.terasologyModules().each {
-        from "$rootDir/modules/${it.name}/build/libs"
-        include "*.jar"
+    // FIXME: duplicating code from /build.gradle:terasologyModules
+    val terasologyModules = rootProject.subprojects.filter { it.parent?.name == "modules" }
+    terasologyModules.forEach {
+        from("$rootDir/modules/${it.name}/build/libs")
+        include("*.jar")
     }
 }
 
-task distPCZip (type: Zip) {
+tasks.register<Zip>("distPCZip") {
     group = "terasology dist"
-    dependsOn distApp
-    dependsOn distModules
-    from "${distsDirectory.get().asFile}/app"
-    archiveFileName = "Terasology.zip"
+    dependsOn("distApp")
+    dependsOn("distModules")
+    from("${distsDirectory.get().asFile}/app")
+    archiveFileName.set("Terasology.zip")
 }
 
-task distForLauncher (type: Sync) {
+tasks.register<Sync>("distForLauncher") {
     group = "terasology dist"
 
-    into rootDirDist
-    from distPCZip
+    into(rootDirDist)
+    from(tasks.getByName("distPCZip"))
 
-    into ("../resources/main/org/terasology/version") {
-        from ("$rootDir/engine/build/classes/org/terasology/version") {
-            include ('versionInfo.properties')
+    into("../resources/main/org/terasology/version") {
+        from("$rootDir/engine/build/classes/org/terasology/version") {
+            include("versionInfo.properties")
         }
     }
 }
 
 // Prep an IntelliJ module for the facade
-idea {
+configure<IdeaModel> {
     module {
         // Change around the output a bit
         inheritOutputDirs = false
-        outputDir = file('build/classes')
-        testOutputDir = file('build/testClasses')
+        outputDir = file("build/classes")
+        testOutputDir = file("build/testClasses")
     }
 }
```

Comment:
Do we still need this with the new idea setup introduced recently?

Reply:
no but no matter how many times people have asked me that none of them have ever removed it :wink:

Reply:
It can affect gestalt source module loading when you compile and run it via idea.
(And it compability with gradle build, it use same output dir, instead standart build/<sourceSet or resource>/classes)

Reply:
facades/PC/ doesn't have modules in it, so for the stuff that applies only to this sub-project we don't have to worry about the gestalt loader.